runtime/port/sysvipc/j9shmem.c

```diff
@@ -1381,6 +1384,22 @@ j9shmem_createDir(struct J9PortLibrary* portLibrary, char* cacheDirName, uintptr
 		}
 	}
 
+	if (0 == omrsysinfo_get_env("HOME", homeDirPathBuffer, J9SH_MAXPATH)) {
+		int pathLen = 1 + strlen(homeDirPathBuffer) + strlen(J9SH_HIDDENDIR);
+		if(pathLen <= J9SH_MAXPATH)
+		{
+			strncpy(cacheDirPathBuffer, homeDirPathBuffer, strlen(homeDirPathBuffer));
+			strncat(cacheDirPathBuffer, "/", 1);
+			strncat(cacheDirPathBuffer, J9SH_HIDDENDIR, strlen(J9SH_HIDDENDIR));
+			if(NULL != strstr(cacheDirName, (const char*)cacheDirPathBuffer))
+			{
```

Comment:
Formatting: space after `if`, opening brace on same line. The cast is not necessary.

Reply:
The test should use `<`, not `<=` (to account for the NUL-terminator).
Suggest this should use `omrstr_printf()` (like above):
```c
		size_t pathLen = strlen(homeDirPathBuffer) + 1 + strlen(J9SH_HIDDENDIR);
		if (pathLen < sizeof(cacheDirPathBuffer)) {
			omrstr_printf(
					cacheDirPathBuffer,
					sizeof(cacheDirPathBuffer),
					"%s/%s",
					homeDirPathBuffer,
					J9SH_HIDDENDIR);
			if (NULL != strstr(cacheDirName, cacheDirPathBuffer)) {
				usingUserHome = TRUE;
			}
```